drivers/input/touchscreen/himax.c

```diff
@@ -0,0 +1,462 @@
+		return error;
+	}
+
+	error = input_register_device(ts->input_dev);
+	if (error) {
+		dev_err(&ts->client->dev, "Failed to register input device: %d",
+			error);
+		return error;
+	}
+
+	return 0;
+}
+
+static u8 himax_event_get_num_points(const struct himax_event *event)
+{
+	if (event->num_points == 0xff)
+		return 0;
+	else
+		return event->num_points & 0x0f;
+}
+
+static bool himax_event_point_is_valid(const struct himax_event_point *point)
+{
+	return point->x != 0xffff && point->y != 0xffff;
+}
+
+static u16 himax_event_point_get_x(const struct himax_event_point *point)
+{
+	return be16_to_cpu(point->x);
+}
+
+static u16 himax_event_point_get_y(const struct himax_event_point *point)
+{
+	return be16_to_cpu(point->y);
+}
+
+static bool himax_process_event_point(struct himax_ts_data *ts,
+				      const struct himax_event *event,
+				      int point_index)
+{
+	const struct himax_event_point *point = &event->points[point_index];
+	u16 x = himax_event_point_get_x(point);
+	u16 y = himax_event_point_get_y(point);
+	u8 w = event->majors[point_index];
+
+	if (himax_event_point_is_valid(point)) {
+		input_mt_slot(ts->input_dev, point_index);
+		input_mt_report_slot_state(ts->input_dev, MT_TOOL_FINGER, true);
+		touchscreen_report_pos(ts->input_dev, &ts->props, x, y, true);
+		input_report_abs(ts->input_dev, ABS_MT_TOUCH_MAJOR, w);
+		input_report_abs(ts->input_dev, ABS_MT_WIDTH_MAJOR, w);
+		return true;
+	} else {
+		return false;
+	}
+}
+
+static void himax_process_event(struct himax_ts_data *ts,
+				const struct himax_event *event)
+{
+	int i;
+	int num_points_left = himax_event_get_num_points(event);
+
+	for (i = 0; i < HIMAX_MAX_POINTS && num_points_left > 0; i++) {
+		if (himax_process_event_point(ts, event, i))
+			num_points_left--;
+	}
+
+	input_mt_sync_frame(ts->input_dev);
+	input_sync(ts->input_dev);
+}
+
+static bool himax_verify_checksum(struct himax_ts_data *ts,
+				  const struct himax_event *event)
+{
+	u8 *data = (u8 *)event;
+	int i;
+	u16 checksum = 0;
+
+	for (i = 0; i < sizeof(*event); i++)
+		checksum += data[i];
+
+	if ((checksum & 0x00ff) != 0) {
+		dev_err(&ts->client->dev, "Wrong event checksum: %04x\n",
+			checksum);
+		return false;
+	} else {
+		return true;
+	}
+}
+
+static void himax_handle_input(struct himax_ts_data *ts)
+{
+	int error;
+	struct himax_event event;
+
+	error = himax_read_input_event(ts, &event);
+	if (error) {
+		dev_err(&ts->client->dev, "Failed to read input event: %d\n",
+			error);
+		return;
+	}
+
+	if (!himax_verify_checksum(ts, &event))
+		return;
+
+	himax_process_event(ts, &event);
+}
+
+static irqreturn_t himax_irq_handler(int irq, void *dev_id)
+{
+	struct himax_ts_data *ts = dev_id;
+
+	himax_handle_input(ts);
+	return IRQ_HANDLED;
+}
+
+static int himax_request_irq(struct himax_ts_data *ts)
+{
+	struct i2c_client *client = ts->client;
+
+	return devm_request_threaded_irq(&client->dev, client->irq, NULL,
+					 himax_irq_handler,
+					 IRQF_TRIGGER_LOW | IRQF_ONESHOT,
```

Comment:
I believe you shouldn't hardcode `IRQF_TRIGGER_LOW` in the driver, as the type from devicetree will be used. So leaving it at only `IRQF_ONESHOT`  should also work, if it's set up in dts correctly